### PR TITLE
Improve error messages for incorrect units in color functions

### PR DESCRIPTION
**Pull request by nex3**
See sass/dart-sass#1745

[skip dart-sass]

### CODE REVIEW

File: spec/core_functions/color/fade_out.hrx [hunk 1]
```diff
@@ -187,3 +187,26 @@ Error: argument `$amount` of `fade-out($color, $amount)` must be between -0 and
...
+<===> error/bounds/unit/error
+Error: $amount: Expected 50% to be within 0 and 1.
+  ,
+3 | a {b: fade-out(red, 50%)}
+  |       ^^^^^^^^^^^^^^^^^^
+  '
+  input.scss 3:7  root stylesheet
+
+<===> error/bounds/unit/error-libsass
+Error: argument `$amount` of `fade-out($color, $amount)` must be between -0 and 1
```
Comment (Goodwine) [inline spec/core_functions/color/fade_out.hrx:207]: Does this mean libsass handled this somewhat correctly already and the tests are just showing what it looks like currently?

File: spec/core_functions/color/change_color/error/bounds.hrx [hunk 2]
```diff
@@ -206,6 +308,39 @@ Error: argument `$lightness` of `change-color($color, $red: false, $green: false
...
 
+<===>
+================================================================================
+<===> lightness/unit/input.scss
+// This test covers sass/dart-sass#1745, but should be removed once units are
+// fully forbidden (sass/sass#3374).
+a {b: change-color(red, $lightness: 200px)}
+
+<===> lightness/unit/error
+DEPRECATION WARNING: $lightness: Passing a number without unit % (200px) is deprecated.
```
Comment (Goodwine) [inline spec/core_functions/color/change_color/error/bounds.hrx:319]: I feel like "a number without unit %" may sound strange to users who may not realize % is considered a unit in sass

Edit: I guess that's out of scope for this change

Comment (nex3) [reply spec/core_functions/color/change_color/error/bounds.hrx:319]: Yeah, I think this is something we just have to expect users to learn at some point.

File: spec/core_functions/color/fade_out.hrx [hunk 3]
```diff
@@ -187,3 +187,26 @@ Error: argument `$amount` of `fade-out($color, $amount)` must be between -0 and
...
+
+<===>
+================================================================================
+<===> error/bounds/unit/input.scss
+// This test covers sass/dart-sass#1745, but should be removed once units are
+// fully forbidden (sass/sass#3374).
+a {b: fade-out(red, 50%)}
+
+<===> error/bounds/unit/error
+Error: $amount: Expected 50% to be within 0 and 1.
```
[the remark this answers is not in the document]
Comment (nex3) [reply spec/core_functions/color/fade_out.hrx:199]: I think that would imply more strongly that passing a number with a unit would also be an error, which isn't true (yet). Eventually we'll have a proper deprecation warning which will be more explicit about this.

File: spec/core_functions/color/fade_out.hrx (hunk 1 above)
Comment (nex3) [reply spec/core_functions/color/fade_out.hrx:207]: Yeah, I guess so!

**Review by Goodwine — APPROVED** [review]

**Pull request merged**